.github/workflows/delegator_generic_slsa3.yml

```diff
@@ -0,0 +1,257 @@
+          slsa-layout-file: ../${{ env.SLSA_OUTPUTS_DIR }}/${{ env.SLSA_ARTIFACTS_FILE }}
+          slsa-workflow-secret1: ${{ secrets.secret1 }}
+          slsa-workflow-secret2: ${{ secrets.secret2 }}
+
+      - name: debug
+        env:
+          OUTPUTS: ${{ toJson(steps.build-artifacts-action.outputs) }}
+        run: |
+          echo "OUTPUTS: $OUTPUTS"
+
+      # NOTE: Needed to upload the file.
+      - name: Move artifact layout file to workspace
+        run: |
+          set -euo pipefail
+          mv "../${{ env.SLSA_OUTPUTS_DIR }}/${{ env.SLSA_ARTIFACTS_FILE }}" "${{ env.SLSA_ARTIFACTS_FILE }}"
+
+      - name: Upload artifact layout file
+        id: upload
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-upload-artifact@main
+        with:
+          name: "${{ needs.rng.outputs.value }}-${{ env.SLSA_ARTIFACTS_FILE }}"
+          path: "${{ env.SLSA_ARTIFACTS_FILE }}"
+
+  # generate-provenance generates and signs the provenance.
+  generate-provenance:
+    needs: [rng, verify-token, privacy-check, build-artifacts-ubuntu]
+    outputs:
+      attestations-download-name: "slsa-attestations-${{ needs.rng.outputs.value }}"
+    permissions:
+      id-token: write # Needed to sign
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download the artifact layout file
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-artifact@main
+        with:
+          name: "${{ needs.rng.outputs.value }}-${{ env.SLSA_ARTIFACTS_FILE }}"
+          path: "${{ env.SLSA_ARTIFACTS_FILE }}"
+          sha256: ${{ needs.build-artifacts-ubuntu.outputs.artifacts-layout-sha256 }}
+
+      - name: Download the predicate file
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-artifact@main
+        with:
+          name: "${{ needs.rng.outputs.value }}-${{ env.SLSA_PREDICATE_FILE }}"
+          path: ${{ env.SLSA_PREDICATE_FILE }}
+          sha256: ${{ needs.verify-token.outputs.predicate-sha256 }}
+
+      - name: debug
+        run: |
+          echo "predicate file: $(cat ${{ env.SLSA_PREDICATE_FILE }})" 
+          echo "artifact file: $(cat ${{ env.SLSA_ARTIFACTS_FILE }})"
+
+      - name: Generate attestations
+        id: attestations
+        uses: laurentsimon/slsa-delegator/.github/actions/generate-attestations@main
+        with:
+          slsa-layout-file: ${{ env.SLSA_ARTIFACTS_FILE }}
+          predicate-type: "https://slsa.dev/provenance/v1.0?draft"
+          predicate-file: ${{ env.SLSA_PREDICATE_FILE }}
+          output-folder: attestations
+
+      - name: Sign attestations
+        id: sign
+        uses: slsa-framework/slsa-github-generator/.github/actions/sign-attestations@main
+        with:
+          attestations: attestations
+          output-folder: slsa-attestations-${{ needs.rng.outputs.value }}
+
+      # NOTE: we don't need to perform a secure upload, since the attestations are signed.
+      - name: Upload attestations
```

Comment:
here you can do the TODO where if you don't have OIDC action you could instead add another job that uploads the output of `attestations` to the folder

Reply:
added a TODO.